vllm/core/scheduler.py

```diff
@@ -386,15 +389,16 @@ def _schedule_running(
         # groups to preempt.
         now = time.time()
         running_queue = policy.sort_by_priority(now, running_queue)
-
         while running_queue:
             seq_group = running_queue[0]
             num_running_tokens = self._get_num_new_tokens(
                 seq_group, SequenceStatus.RUNNING, enable_chunking, budget)
 
-            # We can have up to 1 running prefill at any given time in running
-            # queue, which means we can guarantee chunk size is at least 1.
-            assert num_running_tokens != 0
+            # If chunked prefill is preempted, it is possible chunked prefill
+            # comes before decoding. In that case, it is possible there's no
+            # budget for decoding which returns num_running_tokens == 0.
+            if num_running_tokens == 0:
```

Comment:
We can technically avoid if we always sort waiting queue by FCFS, but we are not doing this because of performance now. 

It can happen in this scenario.

```
step 1
waiting [1, 2, 3]
running []

step 2
waiting []
running [1, 2, 3]

step 3 (all reqs preempted)
waiting [3, 2, 1] # NOTE: we don't sort waiting for perf. preempt is appended to left. 
running []

step 4
waiting [1]
running [2 3] # NOTE: running is always sorted by fcfs

step 5 (1 is scheduled)
waiting []
running [1, 2, 3] ===> 1 is a chunked prefill because it is just scheduled from waiting, and it can consume all token budgets
```